### DISCUSSION TITLE
Difference in calibration between ISIS and OPUS Voyager 2 Images

### DISCUSSION THREAD

**Discussion by JuB104** [DOI-USGS/ISIS3#5216] [Q&A] [answered]
Hi,

I have some .imq images from the Voyager 2 fly-by of Uranus which I am processing using ISIS. I apply _voy2isis_ and _spiceinit_ to them before calibrating them using _voycal_. Now I noticed that the max I/F that results from this process is different from the one that is obtained when using already processed images from the OPUS3 website (https://opus.pds-rings.seti.org/), especially when using the geometrically calibrated 1000x1000 images. 
Is it possible that ISIS' voycal is performing a different radiometric calibration than NASA's Planetary Data System?

Thanks in advance.

**Comment by dpmayerUSGS** [comment, marked as answer]
The derived products available from OPUS, such as geometrically corrected Voyager ISS images were processed by the Multimission Image Processing Lab at JPL, not the PDS (the derived products are also technically non-canonical). MIPL used the VICAR software to create those products using a custom pipeline, so it wouldn't be surprising if the calibrated results from MIPL don't match the results of processing the same raw product in ISIS. It's difficult to say which processing pipeline is "better."

Information about MIPL's workflow is available in the processing notes file in any of the Voyager ISS volumes in OPUS, such as https://opus.pds-rings.seti.org/holdings/volumes/VGISS_7xxx/VGISS_7205/DOCUMENT/PROCESSING.TXT

**Reply by JuB104** [reply]
Thank you very much for the answer @dpmayerUSGS, I appreciate it! 
All the best,
J.

**Reply by JuB104** [reply]
Just a follow-up question: 
MIPL is correcting the geometric distortion of the Voyager cameras by reprojecting it onto a 1000x1000 pixel grid such that the reseau marks fall into their correct spots. Is there any comparable function in ISIS? And is spiceinit, which uses the SPICE kernels to compute ground positions, doing so for the geometrically uncorrected images? 

Thank you in advance.
J.

**Reply by blandoplanet** [reply]
I'm not 100% sure what the MIPL software does, but a similar correction can be performed in ISIS using the _findrx_ application. This uses pattern matching to locate and refine the position of the reseaux. The reseaux position information is then overwritten in the attached cube label and so is then propagated into the geometric distortion correction any time the camera model is used. The change can be significant (depending on the image). To test this, take two copies of the same image, run _findrx_ on one of them and then compare the output from campt. In my view, it's always best practice to run _findrx_ when using Voyager data, since the distortions can be large.

**Reply by dpmayerUSGS** [reply]
I concur with @blandoplanet regarding `findrx`.
`spiceinit` should be run on the geometrically uncorrected images.

**Reply by JuB104** [reply]
Thanks @blandoplanet for the answer, I missed the part in the ISIS documentation where it was explained that _findrx_ corrects the geometric distortion. 

Also Thanks @dpmayerUSGS. So if I understand correctly SPICE does not account for the distortion of Voyager's cameras but produces data for the 'raw' images/ distorted camera model?

**Reply by dpmayerUSGS** [reply]
Ah, we may be using "geometrically uncorrected" to mean different things. `findrx` does not resample or reproject the pixels in the input image. The program updates the label with corrected sample/line locations of reseauxs. If you look at the label of a Voyager ISS image before running `findrx`, you'll see a table of sample/line locations of reseauxs. If you look at the image in `qview`, many of the reseauxs in the image will not appear at the sample/line coordinates reported in the label. `findrx` uses a pattern matching template to try and identify reseauxs in the image and updates the table in the label.

The MIPL processing seems to be different in that they actually warped the original images to force the reseauxs into their nominal positions within a 1000 x 1000 pixel image.

A typical Voyager ISS workflow in ISIS would involve running `voy2isis`, `findrx`, `remrx` (if you want to remove the reseauxs), `spiceinit`, `voycal`, and then `cam2map` if you want to transform an image from the original camera geometry into a map projected image. `spiceinit` will take into account the reseaux location table in the input image, so the results should be more realistic if you run `findrx` before `spiceinit`.

**Reply by JuB104** [reply]
Okay Thanks, that makes sense! I will try out the pipeline you described.
